### PR TITLE
Hook up steady-state `BalanceOfPlant` to `EUDEMO/reactor.py`

### PR DESCRIPTION
**Pull request by CoronelBuendia**
## Description

Adds in an interface to the simplified steady-state balance of plant model in `balance_of_plant` for the EU-DEMO reactor example.

## Interface Changes

Probably some to come

Makes the balance of plant example into an ipynb

## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: tests/builders/EUDEMO/template.json [hunk 1]
```diff
@@ -562,20 +562,13 @@
   "beta_N": {
     "name": "Normalised ratio of plasma pressure to magnetic pressure",
     "value": 2.7,
-<<<<<<< HEAD:tests/builders/EUDEMO/template.json
-    "unit": "dimensionless",
-    "source": "PLASMOD"
-=======
```
Comment (je-cook) [inline tests/builders/EUDEMO/template.json]: It looks like you deleted too much here?

File: tests/builders/EUDEMO/template.json [hunk 2]
```diff
@@ -1974,68 +1947,38 @@
...
   },
   "tk_r_ib_bz": {
     "name": "Thickness ratio of the inboard blanket breeding zone",
     "value": 0.309,
-<<<<<<< HEAD:tests/builders/EUDEMO/template.json
     "unit": "dimensionless",
     "source": "HCPB 2015 design description document"
-=======
-    "unit": "N/A",
-    "source": "Input (HCPB 2015 design description document 2MHDNB)"
```
Comment (je-cook) [inline tests/builders/EUDEMO/template.json]: Wrong source deleted here possibly

**Review by je-cook — COMMENTED** [review]
Conceptually looks fine, l think your json merge conflict commit went a bit wrong

File: tests/builders/EUDEMO/template.json [hunk 3]
```diff
@@ -1006,14 +1218,19 @@
...
+        "send": false
       }
     }
   },
   "m_gas": {
     "name": "Gas puff flow rate",
     "value": 50,
-    "unit": "Pa m^3/s",
-    "source": "Discussions with Chris Day and Yannick H\u00f6rstenmeyer",
+    "unit": "Pam^3/s",
```
Comment (je-cook) [inline tests/builders/EUDEMO/template.json:1232]:
```suggestion
    "unit": "Pa m^3/s",
```

File: tests/builders/EUDEMO/template.json [hunk 4]
```diff
@@ -1368,6 +1609,13 @@
     "value": 0,
     "unit": "ohm",
     "source": "Calculated"
```
Comment (je-cook) [inline tests/builders/EUDEMO/template.json:1611]:
```suggestion
    "source": "Calculated",
```

**Review by je-cook — CHANGES_REQUESTED** [review]
There may be others but I thought this was autogenerated?

**Review by je-cook — APPROVED** [review]
I'm happy here, dunno whether @hsaunders1904 wants this to go in before or after the solvers refactor that he's doing

**Review by hsaunders1904 — APPROVED** [review]
A couple of minor things, but looks good to me otherwise. Happy for it to be merged.

Changes to the solver interface won't be too major, I'll resolve any conflicts that may arise. 

I did spot a `numpy` deprecation warning whilst running one of the examples, although I don't think it was caused by new code in this PR. I'll open a separate ticket to sort that.

**Comment by CoronelBuendia** [comment]
@hsaunders1904 just rebased for now - potentially badly, will pick up comments tomorrow

File: bluemira/builders/EUDEMO/power_cycle.py [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+    def run(self):
+        """
+        Run the setup task.
+        """
+        params = self._params
+        # TODO: Get remaining hard-coded values hooked up
+        neutron_power_strat = NeutronPowerStrategy(
+            f_blanket=0.9,
+            f_divertor=0.05,
+            f_vessel=params.vvpfrac.value,  # TODO: Change this parameter name
```
[the remark this answers is not in the document]
Comment (CoronelBuendia) [reply bluemira/builders/EUDEMO/power_cycle.py:88]: Yes, these are fair comments. My hesitation came from the fact that we are actively discussed how best to manage `Parameter`s that are solely used in one place (i.e. not really globals). Furthermore for these power fractions we technically need a reduced or surrogate neutronics model to determine the neutron power deposited in the various components. I didn't want to bring in my crude volumetric approach prior to the neutronics team having a crack at it.

[the remark this answers is not in the document]
Comment (je-cook) [reply bluemira/builders/EUDEMO/power_cycle.py:88]: Could put it as a note here: https://github.com/Fusion-Power-Plant-Framework/bluemira/projects/21

**Pull request merged**